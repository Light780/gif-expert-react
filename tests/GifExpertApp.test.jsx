import { fireEvent, render, screen } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Pruebas en el <GifExpertApp />', () => {
  test('Debe hace match con el snapshot', () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });

  test('Debe mostrar el titulo GifExpertApp', () => {
    render(<GifExpertApp />);
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toBe(
      'GifExpertApp'
    );
  });

  test('Debe mostrar las categorias', async () => {
    const defaultCategories = ['Dragon Ball', 'One Punch Man'];
    render(<GifExpertApp defaultCategories={defaultCategories} />);
    expect(screen.getAllByRole('heading', { level: 3 }).length).toBe(2);
  });
});
