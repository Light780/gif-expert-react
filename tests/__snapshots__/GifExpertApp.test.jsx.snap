// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifExpertApp /> Debe hace match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
